src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs

```diff
@@ -115,10 +98,27 @@ public MvcRazorHost(ICodeTreeCache codeTreeCache)
 
 #if NET45
         /// <summary>
-        /// The path to the application root.
+        /// Initializes a new instance of <see cref="MvcRazorHost"/> with the specified
+        /// <param name="root"/>.
         /// </summary>
-        public string ApplicationRoot { get; }
+        /// <param name="root">The path to the application base.</param>
+        // Note: This constructor is used by tooling and is created once for each
+        // Razor page that is loaded. Consequently, each loaded page has its own copy of
+        // the CodeTreeCache, but this ok - having a shared CodeTreeCache per application in tooling
+        // is problematic to manage.
+        public MvcRazorHost(string root) :
+            this(new DefaultCodeTreeCache(new PhysicalFileProvider(root)), new DesignTimeRazorPathNormalizer(root))
```

Comment:
nit: we normally keep the colon and constructor call together on one line i.e. `: this(...`